### DISCUSSION TITLE
anyone can train Italian?

### DISCUSSION THREAD

**Discussion by luterz** [SWivid/F5-TTS#535] [General]
this project is very interesting, I don't have a machine powerful enough to train the Italian language, is there anyone who still works with this language? Thank you

**Comment by MithrilMan** [comment]
I'm trying to play with it, I've posted a message here detailing what I tried
https://github.com/SWivid/F5-TTS/discussions/143#discussioncomment-11388813

I still have to test it properly but you can find some info there.
You don't need a machine to run it, you can use runpod or other service to book one but of course it costs

I may release some checkpoints soon on hf even if the model is not great (it won't be great anyway because my test is on italian only so it will forget how to speak in english, I should use a mixed dataset of english/italian to have a multilanguage support)

This is my first attempt at finetuning so I don't expect good results but who knows

Btw would be nice to have an "italian task force" to work on this

**Reply by leoiania** [reply]
Ciao, I'm having the same approach with lot of similarities (italiano anche io!). If you want, we can talk to see if our purposes match and maybe we can help each other.

**Reply by MithrilMan** [reply]
Sure why not, if you have twitter(X) drop me a DM at https://x.com/MithrilMan so we can find a way to have the discussion

**Reply by MithrilMan** [reply]
I've published my test in a public hf repo, don't expect that good results tho :)
https://huggingface.co/alien79/f5-ita-test

trained over the 9h split of facebook multilingual_librispeech without preprocessing for 200 epoch
the file is https://huggingface.co/alien79/f5-ita-test/blob/main/model_italian_200e_9h.safetensors (I've included other checkpoints in the repo)

my settings.json of the gradio interface:

exp_name"F5TTS_Base"
learning_rate0.00001
batch_size_per_gpu4000
batch_size_type"frame"
max_samples64
grad_accumulation_steps1
max_grad_norm1
epochs1053
num_warmup_updates108
save_per_updates2173
last_per_steps200
finetunetrue
file_checkpoint_train""
tokenizer_type"char"
tokenizer_file""
mixed_precision"none"
logger"wandb"
bnb_optimizerfalse

**Reply by MithrilMan** [reply]
to train the custom model with the gradio interface just use this MODEL CKPT:
hf://alien79/f5-ita-test/model_italian_200e_9h.safetensors

**Comment by mikhail2013ru** [comment]
Is there no official support for Italian yet, or is it not important for training a model? How do I create a vocab file for my language correctly? I am interested in the Russian language. I have different datasets, from 5 hours of audio and up to 25. There is a script for kaggle to train the model, but the script needs to be finalized. Write an email, I will send this script. I understand that even without official support of the Russian language, I can train a high-quality model in my own language?
